Only "hardcode" `OAuthImplicitRequest.state` if it was not provided through the config.

schlessera opened the pull request:
There's no reason to simply ignore a value that was set through the config array. If the developer wants to use a specific value (for example, for a non-compliant API), that value should not be silently overridden.

Fixes #13

schlessera committed:
Only "hardcode" values if they were not set through config keys.

There's no reason to simply ignore a value that was set through the config array. If the developer wants to use a specific value (for example, for a non-compliant API), that value should not be silently overridden.

Fixes #13

prayerslayer commented:
> a non-compliant API

My sympathies.

I think for the state it's a reasonable change, but I'm not sure about the grant type values. They are hardcoded because these values are in the RFC. If they are overwritten I'd like to at least emit a warning message.

schlessera commented:
For the `'token'`, I somewhat agree. The preferable way would be to extend `OAuthRequest` with a different type as `OAuthImplicitRequest` in that case. However, if I remember correctly, the base class is not exported, so this is not currently possible.

schlessera commented:
Ah, forgot, same thing applies for `'refresh_token'`. Although I'm not sure a `RefreshRequest` is used in any other context...

schlessera commented:
I suggest changing this pull request to only include de change for the `'state'` for now, and work on a different pull request to export an extendable base class for different `OAuthReuqest` implementations?

prayerslayer commented:
That sounds great! 😃

schlessera committed:
Remove changes to `RefreshRequest.grant_type` and `OAuthImplicitRequest.response_type` for now.

This actually makes sense to hardcode given the requirements in the RFC, and the preferable approach would be to have extendable base classes that can enable different implementations for more exotic needs.

prayerslayer commented:
:+1:

fragsalat commented:
@schlessera Are you working on the base class of OAuthRequest or is this an open task regarding #12 ?

schlessera commented:
@fragsalat I ended up completely changing around the code to fit my needs, so I cannot backport my changes to this repo.

fragsalat commented:
Can you show me the changes? Maybe I can reuse some of them if the issue is still up to date.
